objects/LuaScript.ttslua

```diff
@@ -1470,110 +1490,111 @@ function setupDecks()
 end
 
 function setupMiscThings()
-  -- remove or shuffle bonus encounter cards
-  for i = 1, 5 do
-    local zone = getObjectFromGUID(deckZones["bonus" .. i])
-    for _, obj in ipairs(zone.getObjects()) do
-      if obj.type == "Deck" then
-        if options.expansions.missingExpedition then
-          obj.shuffle()
-        else
-          obj.destruct()
+  if options.campaign == false then
```

Comment:
- This function is never called when campaign is selected
- Nesting almost the whole function in an "if"-statement is bad practice

Reply:
Think this is a remnant from when I was attempting to get the automation of the Red bot player working. I will remove it when I have time.
What is the reasoning for majority of a function being inside an if statement? I've never heard of this being bad practice. But I haven't professionally coded in a year or two.

Reply:
Nesting code like that makes everything afterwards one additional layer in the indentation and thus sucks for readability. Usually, you want to have "early exists" in your if statements

Reply:
Huh interesting! I guess I can see how it might make it more difficult. Not that it’s needed any more but how would you do it without having to create a bunch more code? The first 75% needs to run only when it’s not campaign but the bottom 25% needs to run always.

Reply:
In that case the code should each move into a separate function